aleph-client/src/treasury.rs

```diff
@@ -0,0 +1,164 @@
+use crate::{try_send_xt, wait_for_event, AnyConnection, RootConnection, SignedConnection};
+use ac_primitives::ExtrinsicParams;
+use codec::Decode;
+use frame_support::PalletId;
+use log::info;
+use primitives::Balance;
+use sp_core::{Pair, H256};
+use sp_runtime::{traits::AccountIdConversion, AccountId32};
+use std::{thread, thread::sleep, time::Duration};
+use substrate_api_client::{compose_extrinsic, ApiResult, GenericAddress, XtStatus};
+
+type AnyResult<T> = anyhow::Result<T>;
+
+/// Returns the account of the treasury.
+pub fn treasury_account() -> AccountId32 {
+    PalletId(*b"a0/trsry").into_account_truncating()
+}
+
+/// Returns how many treasury proposals have ever been created.
+///
+/// Requires a single storage read.
+pub fn proposals_counter<C: AnyConnection>(connection: &C) -> u32 {
+    connection
+        .as_connection()
+        .get_storage_value("Treasury", "ProposalCount", None)
+        .expect("Key `Treasury::ProposalCount` should be present in storage")
+        .unwrap_or(0)
+}
+
+/// Calculates how much balance will be paid out to the treasury after each era.
+///
+/// Every call to this method is potentially expensive - it requires three storage reads.
+pub fn staking_treasury_payout<C: AnyConnection>(connection: &C) -> Balance {
+    let sessions_per_era = connection
+        .as_connection()
+        .get_constant::<u32>("Staking", "SessionsPerEra")
+        .expect("Constant `Staking::SessionsPerEra` should be present");
+    let session_period = connection
+        .as_connection()
+        .get_constant::<u32>("Elections", "SessionPeriod")
+        .expect("Constant `Elections::SessionPeriod` should be present");
+    let millisecs_per_block = 2 * connection
+        .as_connection()
+        .get_constant::<u64>("Timestamp", "MinimumPeriod")
+        .expect("Constant `Timestamp::MinimumPeriod` should be present");
+
+    let millisecs_per_era = millisecs_per_block * session_period as u64 * sessions_per_era as u64;
+    let treasury_era_payout_from_staking = primitives::staking::era_payout(millisecs_per_era).1;
+    info!(
+        "[+] Possible treasury gain from staking is {}",
+        treasury_era_payout_from_staking
+    );
```

Comment:
Question: why does this function log? It feels a little weird to me that a getter like this would log... Makes more sense to me that whatever is using this would potentially choose to log the value in its own context. This doesn't seem to be a part of some grand pattern of logging every response either - e.g. `proposals_counter` does not log its output.

Reply:
yeah, ugly leftover - I removed it